### PR TITLE
Splunkpy - Added fetch delta parameter

### PR DESCRIPTION
**Pull request by barchen1**
## Status

- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/14750

## Description
Added the parameter 'fetch time' which allow us to specify how much time back we would like to fetch incidents

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -396,27 +400,29 @@ def splunk_results_command(service):
...
 
-    now = t.strftime(SPLUNK_TIME_FORMAT)
+    now = current_time_for_fetch.strftime(SPLUNK_TIME_FORMAT)
     if demisto.get(demisto.params(), 'useSplunkTime'):
         now = get_current_splunk_time(service)
-        t = datetime.strptime(now, SPLUNK_TIME_FORMAT)
-    if len(lastRun) == 0:
-        t = t - timedelta(minutes=10)
-        lastRun = t.strftime(SPLUNK_TIME_FORMAT)
+        current_time_for_fetch = datetime.strptime(now, SPLUNK_TIME_FORMAT)
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]:
```suggestion
        current_time_in_splunk = datetime.strptime(now, SPLUNK_TIME_FORMAT)
```

File: Integrations/SplunkPy/SplunkPy.py [hunk 2]
```diff
@@ -396,27 +400,29 @@ def splunk_results_command(service):
...
+
+    if len(last_run) == 0:
+        fetch_time_in_minutes = parse_time_to_minutes()
+        current_time_for_fetch = current_time_for_fetch - timedelta(minutes=fetch_time_in_minutes)
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]:
```suggestion
        start_time_for_fetch = current_time_for_fetch - timedelta(minutes=fetch_time_in_minutes)
```

File: Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -438,7 +444,24 @@ def fetch_incidents(service):
...
+    number_of_times, time_unit = FETCH_TIME.split(' ')
+    if str(number_of_times).isdigit():
+        number_of_times = int(number_of_times)
+    else:
+        return_error('Error: Invalid fetch time, need to be a positive integer.')
+    time_unit_value_in_minutes = TIME_IN_MINUTES_DICT.get(time_unit)
+    if time_unit_value_in_minutes:
+        return number_of_times * time_unit_value_in_minutes
+
+    return_error('Error: Invalid time unit.')
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]: use parse_date_range instead, should be more solid solution

File: Integrations/SplunkPy/SplunkPy.py [hunk 4]
```diff
@@ -24,6 +24,10 @@
 VERIFY_CERTIFICATE = not bool(demisto.params().get('unsecure'))
 FETCH_LIMIT = int(demisto.params().get('fetch_limit', 50))
 FETCH_LIMIT = max(min(200, FETCH_LIMIT), 1)
+FETCH_TIME = demisto.params().get('fetch_time')
+TIME_IN_MINUTES_DICT = {'minute': 1, 'minutes': 1, 'hour': 60, 'hours': 60, 'day': 24 * 60, 'days': 24 * 60,
```
Comment (guykeller) [inline Integrations/SplunkPy/SplunkPy.py]: parse_date_range should receive the relevant format and return a datetime object.
You can then parse that datetime to minutes.
What am I missing?

File: Integrations/SplunkPy/SplunkPy.py [hunk 5]
```diff
@@ -438,7 +445,24 @@ def fetch_incidents(service):
...
+def parse_time_to_minutes():
+    """
+    Calculate how much time to fetch back in minutes
+    Returns (int): Time to fetch back in minutes
+    """
+    number_of_times, time_unit = FETCH_TIME.split(' ')
+    if str(number_of_times).isdigit():
+        number_of_times = int(number_of_times)
+    else:
+        return_error('Error: Invalid fetch time, need to be a positive integer.')
```
Comment (guykeller) [inline Integrations/SplunkPy/SplunkPy.py]: '1 minute' is not a positive integer, let's rephrase

File: Integrations/SplunkPy/SplunkPy.py [hunk 6]
```diff
@@ -438,7 +445,24 @@ def fetch_incidents(service):
     if len(incidents) < FETCH_LIMIT:
         demisto.setLastRun({'time': now, 'offset': 0})
     else:
-        demisto.setLastRun({'time': lastRun, 'offset': search_offset + FETCH_LIMIT})
+        demisto.setLastRun({'time': last_run, 'offset': search_offset + FETCH_LIMIT})
+
+
+def parse_time_to_minutes():
```
Comment (guykeller) [inline Integrations/SplunkPy/SplunkPy.py]: feedback on this function - pending reply to previous comment

File: Integrations/SplunkPy/SplunkPy.yml [hunk 7]
```diff
@@ -89,6 +89,13 @@ configuration:
   name: hec_url
   required: false
   type: 0
+- defaultvalue: 3 days
```
Comment (guykeller) [inline Integrations/SplunkPy/SplunkPy.yml]: default value should be 10 minutes to avoid breaking bc

File: Integrations/SplunkPy/SplunkPy.py (hunk 5 above)
Comment (barchen1) [reply Integrations/SplunkPy/SplunkPy.py]: done

File: Integrations/SplunkPy/SplunkPy.yml (hunk 7 above)
Comment (barchen1) [reply Integrations/SplunkPy/SplunkPy.yml]: done

File: Integrations/SplunkPy/SplunkPy.py (hunk 4 above)
Comment (barchen1) [reply Integrations/SplunkPy/SplunkPy.py]: lets talk about it over zoom and ill explain why

File: Integrations/SplunkPy/SplunkPy.py [hunk 8]
```diff
@@ -24,6 +24,10 @@
 VERIFY_CERTIFICATE = not bool(demisto.params().get('unsecure'))
 FETCH_LIMIT = int(demisto.params().get('fetch_limit', 50))
 FETCH_LIMIT = max(min(200, FETCH_LIMIT), 1)
+FETCH_TIME = demisto.params().get('fetch_time')
+TIME_IN_MINUTES_DICT = {'minute': 1, 'minutes': 1, 'hour': 60, 'hours': 60, 'day': 24 * 60, 'days': 24 * 60,
+                        'week': 7 * 24 * 60, 'weeks': 7 * 24 * 60, 'month': 4 * 7 * 24 * 60, 'months': 4 * 7 * 24 * 60,
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]: mark month as 30 or 31 days, and specify this in the docs

Comment (barchen1) [reply Integrations/SplunkPy/SplunkPy.py]: done

File: Integrations/SplunkPy/SplunkPy.py [hunk 9]
```diff
@@ -438,7 +444,27 @@ def fetch_incidents(service):
...
+    Returns (int): Time to fetch back in minutes
+    """
+    number_of_times, time_unit = FETCH_TIME.split(' ')
+    if str(number_of_times).isdigit():
+        number_of_times = int(number_of_times)
+    else:
+        return_error("Error: Invalid fetch time, need to be a positive integer with the time unit afterwards"
+                     " e.g '2 months, 4 days'.")
+    if time_unit[-1] == 's':
+        time_unit = time_unit[:-1]
```
Comment (guykeller) [inline Integrations/SplunkPy/SplunkPy.py]: please add an explaining comment

Comment (barchen1) [reply Integrations/SplunkPy/SplunkPy.py]: done

**Pull request merged**